include/cudaq/Optimizer/Builder/Intrinsics.h

```diff
@@ -86,13 +86,40 @@ class IRBuilder : public mlir::OpBuilder {
   }
 
   cc::GlobalOp
-  genVectorOfComplexConstant(mlir::Location loc, mlir::ModuleOp module,
-                             mlir::StringRef name,
-                             const std::vector<std::complex<double>> &values);
+  genVectorOfConstants(mlir::Location loc, mlir::ModuleOp module,
+                       mlir::StringRef name,
+                       const std::vector<std::complex<double>> &values);
   cc::GlobalOp
-  genVectorOfComplexConstant(mlir::Location loc, mlir::ModuleOp module,
-                             mlir::StringRef name,
-                             const std::vector<std::complex<float>> &values);
+  genVectorOfConstants(mlir::Location loc, mlir::ModuleOp module,
+                       mlir::StringRef name,
+                       const std::vector<std::complex<float>> &values);
+
+  cc::GlobalOp genVectorOfConstants(mlir::Location loc, mlir::ModuleOp module,
+                                    mlir::StringRef name,
+                                    const std::vector<double> &values);
+  cc::GlobalOp genVectorOfConstants(mlir::Location loc, mlir::ModuleOp module,
+                                    mlir::StringRef name,
+                                    const std::vector<float> &values);
+
+  cc::GlobalOp genVectorOfConstants(mlir::Location loc, mlir::ModuleOp module,
+                                    mlir::StringRef name,
+                                    const std::vector<std::int64_t> &values);
+
+  cc::GlobalOp genVectorOfConstants(mlir::Location loc, mlir::ModuleOp module,
+                                    mlir::StringRef name,
+                                    const std::vector<std::int32_t> &values);
+
+  cc::GlobalOp genVectorOfConstants(mlir::Location loc, mlir::ModuleOp module,
+                                    mlir::StringRef name,
+                                    const std::vector<std::int16_t> &values);
+
+  cc::GlobalOp genVectorOfConstants(mlir::Location loc, mlir::ModuleOp module,
+                                    mlir::StringRef name,
+                                    const std::vector<std::int8_t> &values);
+
+  cc::GlobalOp genVectorOfConstants(mlir::Location loc, mlir::ModuleOp module,
```

Comment:
Are there specific tests for all of these?

Reply:
Yes, I had some tests that caught bugs in this code: https://github.com/NVIDIA/cuda-quantum/blob/9b9124c9747f1c11fdfd587b4c2bb68596db4832/targettests/Remote-Sim/args_synthesis.cpp#L1

They are very thorough, so I didn't add anything else.

Reply:
Actually, they are missing cases for synthesizing vectors of complex. Will add tests in another PR.